File: box/src/main/java/de/qabel/box/storage/hash/QabelBoxDigestProvider.java
package de.qabel.box.storage.hash;

import java.security.Provider;
import java.util.UUID;

public class QabelBoxDigestProvider extends Provider {
    public static boolean randomizeName;

    public QabelBoxDigestProvider() {
        super("Qabel" + (randomizeName ? UUID.randomUUID() : ""), 0.1, "Qabel Security Provider");
        put("MessageDigest.Blake2b", Blake2b.class.getName() + "$Digest");
    }
}

File: box/src/main/java/de/qabel/box/storage/hash/QabelBoxDigestProvider.kt
package de.qabel.box.storage.hash

import java.security.Provider

class QabelBoxDigestProvider : Provider("Qabel", 0.1, "Qabel Security Provider") {
    init {
        put("MessageDigest.Blake2b", Blake2b::class.java.name + "\$Digest")
    }
}
